src/metabase/api/collection.clj

```diff
@@ -75,25 +75,35 @@
   the root, if `collection-id` is `nil`) and its immediate children, to avoid reading the entire collection tree when it
   is not necessary.
 
-  For archived, we can either pass in include both (when archived is nil), include only archived is true, or archived is false."
-  [archived exclude-other-user-collections namespace shallow collection-id]
-  (cond->>
-   (t2/select :model/Collection
-              {:where [:and
-                       (when (some? archived)
-                         [:= :archived archived])
-                       (when shallow
-                         (location-from-collection-id-clause collection-id))
-                       (when exclude-other-user-collections
-                         [:or [:= :personal_owner_id nil] [:= :personal_owner_id api/*current-user-id*]])
-                       (perms/audit-namespace-clause :namespace namespace)
-                       (collection/visible-collection-ids->honeysql-filter-clause
-                        :id
-                        (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
-               ;; Order NULL collection types first so that audit collections are last
-               :order-by [[[[:case [:= :type nil] 0 :else 1]] :asc]
-                          [:%lower.name :asc]]})
-    exclude-other-user-collections (remove-other-users-personal-subcollections api/*current-user-id*)))
+  For `archived`, we can either include all values (when `archived` is `nil`),
+  or to include when `archived` is `true` or `false`.
+
```

Comment:
I think the meaning here changed. (It's funny how looking at the code is so much easier to understand sometimes)
```suggestion
For archived, we can either include everthing (when archived is `nil`), only archived (when `archived` is true), or only non-archived (when `archived` is false).
```